src/modules/utils/motordrivercontrol/MotorDriverControl.cpp

```diff
@@ -164,20 +204,29 @@ bool MotorDriverControl::config_module(uint16_t cs)
                 switch(chip) {
                     case DRV8711: drv8711->set_raw_register(&StreamOutput::NullStream, ++reg, i); break;
                     case TMC2660: tmc26x->setRawRegister(&StreamOutput::NullStream, ++reg, i); break;
+                    case TMC2208: tmc22x->setRawRegister(&StreamOutput::NullStream, ++reg, i); break;
                 }
             }
 
             // write the stored registers
             switch(chip) {
                 case DRV8711: drv8711->set_raw_register(&StreamOutput::NullStream, 255, 0); break;
                 case TMC2660: tmc26x->setRawRegister(&StreamOutput::NullStream, 255, 0); break;
+                case TMC2208: tmc22x->setRawRegister(&StreamOutput::NullStream, 255, 0); break;
             }
         }
 
     }else{
         rawreg= false;
     }
 
+    if(chip==TMC2208) {
```

Comment:
no this is incorrect

Reply:
yes, i need a better solution to make sure UART is not used after configuration. I was thinking about deleting the instance after configuration, but then i found this approach..

Reply:
Last commit fixes this and previous one fixes SPI/UART pins issue

Reply:
EDIT: I inserted a new commit moving also SPI and UART objects inside union, can you validate that please